llvm/include/llvm/CodeGen/ValueTypes.td

```diff
@@ -14,6 +14,7 @@
 class ValueType<int size, int value> {
   string Namespace = "MVT";
   string LLVMName = NAME;
+  string LLVMTy = "";
```

Comment:
This is still here, but I guess only used for the scalar values? Could still move this switch into tablegen but I guess that's less important than the vector case

Reply:
This only used for floating point types, part of floating point type can  inference from other fields , but some of them like bf16, ppc f128 and x86 fp80 can't, and bf16 will used by vector types, so I think keep an optional LLVMTy field should be easier to maintain here.

Reply:
Oh, I guess I van just inference from the name.